main/src/ca/uwaterloo/flix/language/phase/Inliner.scala

```diff
@@ -297,7 +297,8 @@ object Inliner {
     * Checks if `occur` is Dead and purity is `Impure`
     */
   private def isDeadAndImpure(occur: OccurrenceAst.Occur, purity: Purity): Boolean = (occur, purity) match {
```

Comment:
@JonathanStarup I am not sure where this function is used, but this seems dangerous to me. Would this not end up removing expressions with control-effects?

Reply:
Its just the wording thats misleading. I've changed it to `isDeadAndNotPure` now.